consensus/istanbul/validator/default.go

```diff
@@ -81,52 +92,29 @@ type defaultSet struct {
 	randomness common.Hash
 }
 
-type defaultSetRLP struct {
-	Validators []*defaultValidator
-	Randomness common.Hash
-}
-
-func (val *defaultSet) DecodeRLP(stream *rlp.Stream) error {
-	var v defaultSetRLP
-	if err := stream.Decode(&v); err != nil {
-		return err
-	}
-
-	validators := make([]istanbul.Validator, len(v.Validators))
-	for i := range v.Validators {
-		validators[i] = v.Validators[i]
-	}
-
-	*val = defaultSet{
-		validators: validators,
-		randomness: v.Randomness,
+func newDefaultSet(validators []istanbul.ValidatorData) *defaultSet {
+	return &defaultSet{
+		validators: mapDataToValidators(validators),
 	}
-	return nil
-}
-
-func (val *defaultSet) EncodeRLP(w io.Writer) error {
-	return rlp.Encode(w, []interface{}{
-		val.validators,
-		val.randomness,
-	})
 }
 
-func (val *defaultSet) Serialize() ([]byte, error) {
-	return rlp.EncodeToBytes(val)
+func NewValidatorSetFromData(data *istanbul.ValidatorSetData) istanbul.ValidatorSet {
+	return newValidatorSetFromData(data)
 }
 
-func newDefaultSet(validators []istanbul.ValidatorData) *defaultSet {
-	valSet := &defaultSet{}
-
-	// init validators
-	valSet.validators = make([]istanbul.Validator, len(validators))
-	for i, validator := range validators {
-		valSet.validators[i] = New(validator.Address, validator.BLSPublicKey)
+func newValidatorSetFromData(data *istanbul.ValidatorSetData) *defaultSet {
+	return &defaultSet{
+		validators: mapDataToValidators(data.Validators),
+		randomness: data.Randomness,
```

Comment:
I don't think we should create this function, especially because it is unused.

Reply:
Removed!